subsystem/kubeapi_test.go

```diff
@@ -644,18 +646,77 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		Eventually(func() string {
 			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.ISODownloadURL
 		}, "15s", "5s").Should(Not(BeEmpty()))
+		// adding sleep to be sure  that the backend function GenerateClusterISOInternal() will not reject the
+		// request with "request came too soon after previous request"
+		time.Sleep(2 * bminventory.WindowBetweenRequestsInSeconds)
 
 		infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName)
 		firstURL := infraEnv.Status.ISODownloadURL
+		firstCreatedAt := infraEnv.Status.CreatedAt
 
+		annotations := map[string]string{"foo": "bar"}
+		By("Update InfraEnv Annotations")
 		Eventually(func() error {
-			infraEnv.SetAnnotations(map[string]string{"foo": "bar"})
+			infraEnv.SetAnnotations(annotations)
 			return kubeClient.Update(ctx, infraEnv)
 		}, "30s", "10s").Should(BeNil())
 
+		Consistently(func() map[string]string {
+			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).GetAnnotations()
+		}, "10s", "2s").Should(Equal(annotations))
+
+		By("Verify InfraEnv Status ISODownloadURL has not changed")
 		Consistently(func() string {
 			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.ISODownloadURL
 		}, "10s", "2s").Should(Equal(firstURL))
+
+		By("Verify InfraEnv Status CreatedAt has not changed")
+		Consistently(func() *metav1.Time {
+			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.CreatedAt
+		}, "10s", "2s").Should(Equal(firstCreatedAt))
+	})
+
+	It("verify InfraEnv ISODownloadURL and image CreatedAt are changing  - update IgnitionConfigOverride", func() {
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		deployAgentClusterInstallCRD(ctx, kubeClient, aciSpec, clusterDeploymentSpec.ClusterInstallRef.Name)
+		By("Deploy InfraEnv")
+		deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+
+		infraEnvKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      infraNsName.Name,
+		}
+
+		Eventually(func() string {
+			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.ISODownloadURL
+		}, "15s", "5s").Should(Not(BeEmpty()))
+		// adding sleep to be sure  that the backend function GenerateClusterISOInternal() will not reject the
+		// request with "request came too soon after previous request"
+		time.Sleep(2 * bminventory.WindowBetweenRequestsInSeconds)
```

Comment:
not sure if it's needed, you are checking eventually and consistently so why do you need this wait?

Reply:
same as previous response